homeassistant/components/climate/zhong_hong.py

```diff
@@ -0,0 +1,179 @@
+        self.schedule_update_ha_state()
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self.unique_id
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the HVAC."""
+        return "zhong_hong_hvac_{}_{}".format(self._device.addr_out,
+                                              self._device.addr_in)
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return (SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+                | SUPPORT_OPERATION_MODE | SUPPORT_ON_OFF)
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._device.current_operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._device.operation_list
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._device.current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._device.target_temperature
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self._device.is_on
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        return self._device.current_fan_mode
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return self._device.fan_list
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return convert_temperature(self._device.min_temp, TEMP_CELSIUS,
+                                   self.temperature_unit)
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return convert_temperature(self._device.max_temp, TEMP_CELSIUS,
+                                   self.temperature_unit)
+
+    def turn_on(self):
+        """Turn on ac."""
+        return self._device.turn_on()
+
+    def turn_off(self):
+        """Turn off ac."""
+        return self._device.turn_off()
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is not None:
+            self._device.set_temperature(temperature)
+
+        operation_mode = kwargs.get(ATTR_OPERATION_MODE)
+        if operation_mode is not None:
```

Comment:
mode is not part of setting a temperature?

Reply:
It is.

Reply:
operation_mode is passed in at here https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/climate/__init__.py#L190